### PR TITLE
Plug leaks

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

**Comment by Therzok** [comment]
https://gist.github.com/Therzok/709f0e6088b6e0af5bc03a833c184b00

**Comment by Therzok** [comment]
cc @DavidKarlas @alanmcgovern

File: gtk/Window.custom [hunk 1]
```diff
@@ -59,11 +58,10 @@
 				IntPtr raw_ret = gtk_window_get_icon_list(Handle);
 				if (raw_ret == IntPtr.Zero)
 					return new Gdk.Pixbuf [0];
-				GLib.List list = new GLib.List(raw_ret);
-				return GLib.Marshaller.ListToArray<Gdk.Pixbuf> (list);
+				return GLib.Marshaller.ListPtrToArray<Gdk.Pixbuf> (raw_ret, typeof (GLib.List), true, false);
 			}
 			set {
-				GLib.List list = new GLib.List(IntPtr.Zero);
+				GLib.List list = new GLib.List(IntPtr.Zero, typeof (IntPtr), true, false);
```
Comment (alanmcgovern) [inline gtk/Window.custom]: should we dispose this immediately?

Comment (Therzok) [inline gtk/Window.custom]: It's going to free everything on finalization.

File: gtk/TextIter.custom [hunk 2]
```diff
@@ -67,8 +65,7 @@
 			IntPtr raw_ret = gtk_text_iter_get_toggled_tags (ref this, toggled_on);
 			if (raw_ret == IntPtr.Zero)
 				return new TextTag [0];
-			GLib.SList list = new GLib.SList(raw_ret);
-			return GLib.Marshaller.ListToArray<TextTag> (list);
+				return GLib.Marshaller.ListPtrToArray<TextTag> (raw_ret, typeof (GLib.SList), true, false);
```
Comment (alanmcgovern) [inline gtk/TextIter.custom]: bad indenting?

File: gtk/Window.custom (hunk 1 above)
Comment (alanmcgovern) [inline gtk/Window.custom]: free :allthethings: ;p

Thats ok :)

**Comment by alanmcgovern** [comment]
lgtm

File: gtk/TextIter.custom (hunk 2 above)
Comment (Therzok) [inline gtk/TextIter.custom]: Yeah, but doesn't matter that much.

Comment (alanmcgovern) [inline gtk/TextIter.custom]: my poor eyes :(

:P

**Comment by bratsche** [comment]
How much of this applies to gtk#3 / master?

**Comment by Therzok** [comment]
A big part, I guess? I'd need to find out if gtk breaks annotation contract between versions. If not, then all of this applies to master.

**Pull request merged**